CRM/Core/Payment/BaseIPN.php

```diff
@@ -270,18 +271,18 @@ public function cancelled($objects) {
           }
         }
       }
+      $participant = &$objects['participant'];
+
+      if ($participant) {
+        $this->cancelParticipant($participant->id);
+      }
     }
     else {
       Contribution::update(FALSE)->setValues([
         'cancel_date' => 'now',
         'contribution_status_id:name' => 'Cancelled',
       ])->addWhere('id', '=', $contribution->id)->execute();
     }
-    $participant = &$objects['participant'];
```

Comment:
just moves this in the loop of unreachable code - same as memberships - not really needed now that the whole function is no longer called I guess